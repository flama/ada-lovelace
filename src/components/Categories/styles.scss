@import '../../config';

.categories {

  .title {
    display: block;
    margin-bottom: $space-l;
  }

  .main {
    margin-bottom: $space-m;

    .button {
      padding-left: $space-s;
      padding-right: $space-s;
      margin-left: $space-xxs;

      border-style: solid;
      border-width: 1px;
      border-color: $light-purple;
      border-radius: 0.5em;

      font-size: $font-size-centi;
      line-height: 1.5em;
      letter-spacing: 0.44px;
      color: $light-purple;
      background-color: transparent;
      cursor: pointer;

      transition: all 0.3s ease;

      &:first-child { margin-left: 0; }

      &:hover { color: #fff; }

      &.selected {
        border-color: transparent;
        background-color: $selected;
        color: #fff;
      }
    }
  }
}
