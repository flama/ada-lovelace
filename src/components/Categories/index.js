import React, { Component } from 'react'

import './styles.scss'

const ALL = -1

class Categories extends Component {

  constructor(props) {
    super(props)

    this.state = {
      selected: ALL
    }
  }

  styleTitle = position => {
    return `title -${position}`
  }

  styleButton = index => {
    return `button ${index === this.state.selected? 'selected':''}`
  }

  choose = selected => {
    this.setState({ selected })
    this.props.onChange(selected)
  }

  render() {
    let options = this.props.options.map((option, index) => {
      return (
        <button className={ this.styleButton(index) }
          key={ option }
          onClick={ () => this.choose(index)}
        >
          { option }
        </button>
      )
    })

    return (
      <div className="categories">
        <div className="content">
          <h4 className={ this.styleTitle(this.props.titlePosition) }>
            Who are the wonderful women who helped shape the world?
          </h4>
          <div className="legend">
            <span>Women</span>
          </div>
        </div>
        <div className="main">
          <button className={ this.styleButton(ALL) }
            onClick={ () => this.choose(ALL) }
          >
            { this.props.all }
          </button>
          { options }
        </div>
      </div>
    )
  }
}

Categories.defaultProps = {
  onChange: _=>_,
  options: [],
  all: "All",
  titlePosition: "inline"
}

Categories.propTypes = {
  onChange: React.PropTypes.func,
  title: React.PropTypes.string.isRequired,
  options: React.PropTypes.arrayOf(React.PropTypes.string),
  all: React.PropTypes.string,
  titlePosition: React.PropTypes.oneOf(['top', 'inline']),
}

export default Categories
