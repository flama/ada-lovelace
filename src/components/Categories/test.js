import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import Categories from '../Categories'

describe('Categories', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Categories />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
