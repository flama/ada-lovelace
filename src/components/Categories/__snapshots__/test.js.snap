// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories renders correctly 1`] = `
<div
  className="categories"
>
  <button
    className="button selected"
    onClick={[Function]}
  >
    All
  </button>
  <button
    className="button"
    id="option1"
    onClick={[Function]}
  >
    option1
  </button>
  <button
    className="button"
    id="option2"
    onClick={[Function]}
  >
    option2
  </button>
  <button
    className="button"
    id="testoption"
    onClick={[Function]}
  >
    testoption
  </button>
</div>
`;
